pkg/generate/app/pipeline.go

```diff
@@ -140,24 +141,24 @@ func (g PipelineGroup) String() string {
 	return strings.Join(s, "+")
 }
 
-const MaxServiceNameLen = 24
+const maxServiceNameLength = 24
 
-var InvalidServiceChars = regexp.MustCompile("[^-a-z0-9]")
+var invalidServiceChars = regexp.MustCompile("[^-a-z0-9]")
 
-func makeValidServiceName(name string) string {
-	name = strings.ToLower(name)
-	name = InvalidServiceChars.ReplaceAllString(name, "")
-	if len(name) == 0 {
-		return fmt.Sprintf("svc-%d", rand.Intn(100000))
-	}
-	if len(name) > MaxServiceNameLen-5 {
-		name = name[:MaxServiceNameLen-5]
+func makeValidServiceName(name string) (string, string) {
+	if ok, _ := validation.ValidateServiceName(name, false); ok {
+		return name, ""
```

Comment:
The problem I ran into with builders like the ruby-20-centos that have 2 (8080,8443) is that sometimes I'd get 8080 first and other times 8443 and it was non-deterministic because they were coming from a map